prov/shm/src/smr_progress.c

```diff
@@ -322,6 +323,13 @@ static int smr_progress_iov(struct smr_cmd *cmd, struct iovec *iov,
 		goto out;
 	}
 
+	xpmem = &smr_peer_data(ep->region)[cmd->msg.hdr.id].xpmem;
+	if (ep->region->xpmem_cap_self == SMR_VMA_CAP_ON &&
+	    xpmem->cap == SMR_VMA_CAP_ON) {
+		p2p_type = FI_SHM_P2P_XPMEM;
```

Comment:
can we set the p2p_type earlier when we initialize the ep so we don't have to check this every time on the fast path?